tools-image/enki/pkg/config/config.go

```diff
@@ -0,0 +1,295 @@
+	}
+
+	// Set formatter so both file and stdout format are equal
+	log.SetFormatter(&logrus.TextFormatter{
+		ForceColors:      true,
+		DisableColors:    false,
+		DisableTimestamp: false,
+		FullTimestamp:    true,
+	})
+
+	// Logfile
+	logfile := viper.GetString("logfile")
+	if logfile != "" {
+		o, err := vfs.OpenFile(logfile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, fs.ModePerm)
+
+		if err != nil {
+			log.Errorf("Could not open %s for logging to file: %s", logfile, err.Error())
+		}
+
+		// else set it to both stdout and the file
+		mw := io.MultiWriter(os.Stdout, o)
+		log.SetOutput(mw)
+	} else { // no logfile
+		if viper.GetBool("quiet") { // quiet is enabled so discard all logging
+			log.SetOutput(io.Discard)
+		} else { // default to stdout
+			log.SetOutput(os.Stdout)
+		}
+	}
+
+	v := version.Get()
+	if log.GetLevel() == logrus.DebugLevel {
+		log.Debugf("Starting enki version %s on commit %s", v.Version, v.GitCommit)
+	} else {
+		log.Infof("Starting enki version %s", v.Version)
+	}
+}
+
+// BindGivenFlags binds to viper only passed flags, ignoring any non provided flag
+func bindGivenFlags(vp *viper.Viper, flagSet *pflag.FlagSet) {
+	if flagSet != nil {
+		flagSet.VisitAll(func(f *pflag.Flag) {
+			if f.Changed {
+				_ = vp.BindPFlag(f.Name, f)
+			}
+		})
+	}
+}
+
+// setDecoder sets ZeroFields mastructure attribute to true
+func setDecoder(config *mapstructure.DecoderConfig) {
+	// Make sure we zero fields before applying them, this is relevant for slices
+	// so we do not merge with any already present value and directly apply whatever
+	// we got form configs.
+	config.ZeroFields = true
+}
+
+type Unmarshaler interface {
+	CustomUnmarshal(interface{}) (bool, error)
+}
+
+func UnmarshalerHook() mapstructure.DecodeHookFunc {
+	return func(from reflect.Value, to reflect.Value) (interface{}, error) {
+		// get the destination object address if it is not passed by reference
+		if to.CanAddr() {
+			to = to.Addr()
+		}
+		// If the destination implements the unmarshaling interface
+		u, ok := to.Interface().(Unmarshaler)
+		if !ok {
+			return from.Interface(), nil
+		}
+		// If it is nil and a pointer, create and assign the target value first
+		if to.IsNil() && to.Type().Kind() == reflect.Ptr {
+			to.Set(reflect.New(to.Type().Elem()))
+			u = to.Interface().(Unmarshaler)
+		}
+		// Call the custom unmarshaling method
+		cont, err := u.CustomUnmarshal(from.Interface())
+		if cont {
+			// Continue with the decoding stack
+			return from.Interface(), err
+		}
+		// Decoding finalized
+		return to.Interface(), err
+	}
+}
+
+func viperReadEnv(vp *viper.Viper, prefix string, keyMap map[string]string) {
+	// If we expect to override complex keys in the config, i.e. configs
+	// that are nested, we probably need to manually do the env stuff
+	// ourselves, as this will only match keys in the config root
+	replacer := strings.NewReplacer("-", "_")
+	vp.SetEnvKeyReplacer(replacer)
+
+	if prefix == "" {
+		prefix = "ELEMENTAL"
```

Comment:
No need to use "elemental" anywhere anymore no? We can do the replacement later but it would be better if we don't introduce it at all. Is there any code out there that expects this prefix? Do we need a deprecation plan?

Reply:
nope, this is a side effect of bringing the full thing, this should not be needed anymore as we dont pass env vars to override the configs.